features/FEATURE_BLE/ble/Gap.h

```diff
@@ -82,44 +87,84 @@ class GapAdvertisingData;
  * sending a scan request. If the broadcaster accepts scan requests, it can reply
  * with a scan response packet containing additional information.
  *
- * @code
- * // assuming gap has been initialized
- * Gap& gap;
+ * Advertising parameters are updated using setAdvertisingParams(). The main
+ * advertising payload is updated using setAdvertisingPayload() and the scan response
+ * is updated using setAdvertisingScanResponse(). If the advertising is already active
+ * updated the data will take effect from the next advertising event.
  *
- * // construct the packet to advertise
- * GapAdvertisingData advertising_data;
+ * To create a valid advertising payload and scan response you may use
+ * AdvertisingDataBuilder(). You must first allocate memory and crate an mbed::Span and
+ * pass that into the AdvertisingDataBuilder which will only be able to add as much
+ * data as fits in the provided buffer. The builder will accept any size of the buffer
+ * but for the created data to be usable it must be smaller than the maximum data
+ * length returned from getMaxAdvertisingDataLength().
  *
- * // Add advertiser flags
- * advertising_data.addFlags(
- *    GapAdvertisingData::LE_GENERAL_DISCOVERABLE |
- *    GapAdvertisingData::BREDR_NOT_SUPPORTED
- * );
+ * @note The maximum size of data depends on the controller and its support for
+ * extended advertising however even if the controller supports larger data lengths if
+ * you wish to be compatible with older devices you may wish to use legacy
+ * advertising and should not use payloads larger than LEGACY_ADVERTISING_MAX_SIZE.
  *
```

Comment:
Might be worth mentioning that this feature was introduced in Bluetooth 5 and that 5 support from the controller's standpoint is necessary but not sufficient.

Reply:
@donatieng I thin I addressed your comment.